main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1322,6 +1322,26 @@ object Weeder {
           mkApplyFqn("List.append", List(e1, e2), loc)
       }
 
+    case ParsedAst.Expression.FList(sp1, sp2, exps) =>
+      /*
+       * Rewrites a `FList` expression into `List.Nil` with `List.Cons`.
+       */
+      val loc = mkSL(sp1, sp2).asSynthetic
+
+      traverse(exps)(e => visitExp(e, senv)) map {
+        case es =>
+          val nilName = Name.Ident(sp1, "Nil", sp2).asSynthetic
```

Comment:
@mlutze Is this right? Both with the use of the ident and with synthetic?

Reply:
We probably want it to be a `List.Nil`

Reply:
Isn't that covered by the `tag` part?

Reply:
Yes, my bad

Reply:
What about synth, is it ok?

Reply:
Looks OK to me